### DISCUSSION TITLE
ChatBot

### DISCUSSION THREAD

**Discussion by Arslanzeyakhan** [community/community#48999] [Programming Help]
**Chatbot Python**

Question

### Body

Hello there, This is Arslan from a different side of the world and i want you to help me pull this off this is an project where we create Chat bot , Well bascially just improve it 
![image](https://user-images.githubusercontent.com/74643255/222492991-81b927f2-81aa-4324-9c54-2059874e0596.png)
Because as you can see i already have completed the first two tasks which were meant to basically accomplish the Goal to end the stereotype that there is an separate language for creating a Chat bot , But after Going through like 100 bugs i accomplished the goal within 6 months now my second concern is raise when i want to create an project which could make an Chat bot already securing a pile of code in Python to be modified to its hill so we can finally come Out to the community pointing towards the special debates thereafter 
Can you Help me pull this off ??  if you want to reach up to me as an Helping Hand this is my Discord ID :- Arslanhack13#4095 feel free to reach to me for any query RELATED TO the topic otherwise you can ask me things which are IN GERENAL in the COMMENT section and also if you feel that you have noticed a question which can be raised in a more Confident way then feel free to reach me on discord Arslanhack13#4095  
Cheers
Arslan

**Comment by Arslanzeyakhan** [comment]
If you can / have a good type of a experience with bots Strictly Pointing towards the field of Python i will love if you will engage in this project light weight this is on IDE :- VS code 2021
Cheers
Arslan

**Reply by ghost** [reply]
Greetins , @Arslanzeyakhan
we have different chatbots for  different purposes 
i'll mention some of them
for example you can have these chatbots and their proper libraries to 
construct them.

- discord chatbots -> **discord.py (search pypi.org)**

- telegram chatbots -> **telethon , python-telegram-bot , pyrogram**

- network-based chatbots -> **socket , twisted** 

etc.

each of them has it's own documentation across the internet and you can surely 
find a good one.

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @Arslanzeyakhan ! 

We’ve moved your post to our new Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.

**Comment by DANLOANS** [comment]
I recommend you to use discord.js because it's easier.

**Comment by KauaAmaroPires** [comment]
Good afternoon, I liked the project idea, but I'd like to suggest something. Instead of using fixed expected text in the user's question, how about using regular expressions (regex) designed to detect keywords within the words? I believe you could save a lot of code and variations of the same question, you understand? For example: "What is your name?" The regex would capture only the keywords: "What" and "name" and provide the answer.

**Comment by AndresGoGutierrez** [comment]
Hi Arslan, great job on completing the first tasks! I think your idea of improving the chatbot could be taken further by structuring the Python code in a more modular way.
